charts/opentelemetry-operator/README.md

```diff
@@ -0,0 +1,319 @@
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: []
+          exporters: [logging]
+EOF
+```
+
+### DaemonSet Mode
+
+DaemonSet should satisfy your most basic needs and is the most common method to deploy OpenTelemetry collector.
+In this case, every Kubernetes node will have its own collector copy which would monitor the pods in it.
+
+The following example configuration deploys the Collector as DaemonSet resource. The receiver is jaeger-receiver and
+the exporter is logging exporter.
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: my-collector
+spec:
+  mode: daemonset
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+    processors:
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: []
+          exporters: [logging]
+EOF
+```
+
+### StatefulSet Mode
+If you want your collector to have stable persistent identities or storage, you should choose StatefulSet.
+Take Prometheus metrics for example, if you use above two approaches, the metrics collected from the receiver will
+be stored locally by default, which is ephemeral. However, StatefulSet allows you to configure a persistent storage
+interface called PersistentVolume. PersistentVolume will maintain the historical metrics data and even survive pods
+restart. This feature gives you the possibility that you can reconstruct the time-series logs.
+
+The following example configuration deploys the Collector as StatefulSet resource with three replicas. The receiver
+is jaeger-receiver and the exporter is logging exporter.
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: my-collector
+spec:
+  mode: statefulset
+  volumeMounts:
+    - mountPath: "/usr/share/test-volume"
+      name: test-volume
+  volumeClaimTemplates:
+    - metadata:
+        name: "test-volume"
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        storageClassName: "standard"
+        resources:
+          requests:
+            storage: 1Gi
+  replicas: 3
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+    processors:
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: []
+          exporters: [logging]
+EOF
+```
+
+### Sidecar Mode
+If all you want to monitor is a single application, then collector as the sidecar would be the best fit.
+This collector instance will work on the container level and no new pods or other workload resources are needed,
+which is perfect to keep your Kubernetes cluster clean. Moreover, you can also use the sidecar mode when you want
+to use a different collect/export strategy, which just suits this application.
```

Comment:
While this is certainly a valid case, I wouldn't say this is the biggest use case. This mode exists for people to offload their telemetry data as fast and reliable as possible from their applications. A sidecar (ie, container in the same pod) is good for that, as it's on the same network as the application's process. 

This sidecar would typically send the telemetry data to a remote OpenTelemetry Collector, either as a stateful set or deployment, which is then configured to send data to a final destination. If you decide to switch your Jaegers or to send data to a new vendor, you can just change the collector configuration in the deployment. Changing the sidecars would require all the pods to be recreated.

Reply:
> This mode exists for people to offload their telemetry data as fast and reliable as possible from their applications. A sidecar (ie, container in the same pod) is good for that, as it's on the same network as the application's process.

Will add this to the README!